bpf/lib/conntrack.h

```diff
@@ -917,6 +929,9 @@ ct_create_fill_entry(struct ct_entry *entry, const struct ct_state *state,
 
 	if (dir == CT_SERVICE) {
 		entry->backend_id = state->backend_id;
+#ifdef ENABLE_ACTIVE_CONNECTION_TRACKING
+		_lb_act_conn_open(state->rev_nat_index, state->backend_id);
+#endif
 	} else if (dir == CT_INGRESS || dir == CT_EGRESS) {
```

Comment:
Picking on this part in particular, as it's clearly misplaced. `ct_create_fill_entry()` is meant to fill the CT entry (from the info in the `ct_state` representation), but here we don't even touch the CT entry. So at best this part should just live in `ct_create{4,6}()` instead.

But as more general question - why are these changes intertwined with the actual CT lookup? If we call the feature `Active connection counting` instead, would it still fit into `conntrack.h`? We're not actually tracking any connection state, after all ...

My expectation would be to have this logic live [here](https://github.com/cilium/cilium/blob/32e76ffb401d94b40307d281512606f21fc71f1e/bpf/lib/lb.h#L1525), where it would be included into the BPF program only **once**. What's missing to make that happen? Do we just need to pass back a few bits of connection-internal state to detect reopened / closed connections?

Reply:
>  it's clearly misplaced. ct_create_fill_entry() is meant to fill the CT entry

Agreed. I moved all calls to a single function, `__ct_lookup`.

> If we call the feature Active connection counting instead, would it still fit into conntrack.h?

Yes, the decision to place it in `conntrack.h` is a technical requirement of what information is available in each function. Please, feel free to suggest other places that are suitable for this, if there are any.

> My expectation would be to have this logic live [here](https://github.com/cilium/cilium/blob/32e76ffb401d94b40307d281512606f21fc71f1e/bpf/lib/lb.h#L1525)

This handles counting new connections only, where do you want to handle closing connections?

Reply:
>what information is available in each function. Please, feel free to suggest other places that are suitable for this, if there are any.

Let's use the `ct_state` then to provide this detailed level of information to callers :).

>This handles counting new connections only, where do you want to handle closing connections?

How about the following?

```
ct_result = ct_lazy_lookup4(...);
if (ct_result == CT_NEW || ct_state->reopened)
	// account for opened connection
else if (ct_state->closing)
	// account for closed connection
```